src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const dateParts = {};
+  const dateValues = date.split(fromSeparator);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateParts[fromFormat[i]] = dateValues[i];
+  }
+
+  if (dateParts['YY']) {
+    const year = parseInt(dateParts['YY'], 10);
+
+    if (year < 30) {
+      dateParts['YYYY'] = `20${dateParts['YY']}`;
+    } else {
+      dateParts['YYYY'] = `19${dateParts['YY']}`;
+    }
+  }
+
+  if (dateParts['YYYY'] && !dateParts['YY']) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+  }
+
+  const formattedDate = toFormat
+    .slice(0, 3)
+    .map((part) => dateParts[part])
+    .join(toSeparator);
```

Comment:
The use of 'slice(0, 3)' assumes that the 'toFormat' array will always have the date parts in the first three positions, which may not be the case. Instead, consider iterating over the 'toFormat' array, excluding the separator, and build the formatted date based on the presence of each part in the 'dateParts' object.